packages/components/src/internal/components/entities/EntityInsertPanel.tsx

```diff
@@ -941,6 +960,72 @@ class EntityInsertPanelImpl extends Component<Props, StateProps> {
         );
     };
 
+    static getWarningFieldList(names: string[])  {
+        const oxfordComma = names.length > 2 ? ',' : '';
+        return names.map((name, index) => (
+            <span key={name}>
+                <b>{name}</b>{index === names.length-2 ? oxfordComma + ' and ' : index < names.length-2 ? ', ': ''}
+            </span>
+        ))
+    }
+
+    static getInferredFieldWarnings(inferred: InferDomainResponse, domainDesign: DomainDesign, columns: OrderedMap<string, QueryColumn>) : Array<ReactNode> {
+        let uniqueIdFields = [];
+        let unknownFields = [];
+
+        inferred.fields.forEach(field => {
+            const lcName = field.name.toLowerCase();
+            let aliasField = domainDesign.fields.find(domainField => domainField.importAliases?.toLowerCase().indexOf(lcName) >= 0);
+            const columnName = aliasField ? aliasField.name : field.name;
+            const column = columns.find(column => (column.isImportColumn(columnName)));
+
+            if (!column) {
+                if (unknownFields.indexOf(field.name) < 0) {
+                    unknownFields.push(field.name);
+                }
+            }
+            else if (column.isUniqueIdColumn) {
+                if (uniqueIdFields.indexOf(field.name) < 0) { // duplicate fields are
```

Comment:
looks like the comment here was truncated

Reply:
Fixed